lib/binary_tree.c

```diff
@@ -0,0 +1,42 @@
+#include <stdlib.h>
+
+#include "binary_tree.h"
+
+struct bt_node *bt_new_node(int val)
+{
+    struct bt_node *bt_node_ptr = malloc(sizeof(struct bt_node));
+    if (bt_node_ptr) {
+        bt_node_ptr->data = val;
+        bt_node_ptr->left = bt_node_ptr->right = NULL;
+    }
+    return bt_node_ptr;
+}
+
+void bt_insert(struct bt_node **node, int val)
+{
+    if (*node == NULL)
+        *node = bt_new_node(val);
```

Comment:
Dangerous call! Let the caller know about possible failure in bt_node_ptr() or just ASSERT().